### PR TITLE
fix(react): Remove transient dependencies from direct dependencies.

### PR DESCRIPTION
**Pull request by cg-sat**
Proposal 1 for #114 
@ibgreen I'm not familiar with the projects' policy how to maintain transient dependencies. If proposal 2 is desired, I can change the PR accordingly.

### CODE REVIEW

**Comment by cg-sat** [comment]
I just saw, that the code actually imports stuff from `styled-components` and `react`. This means, that these two packages are not only transient dependencies but direct dependencies. This means they should also be made explicit.

File: modules/react/package.json [hunk 1]
```diff
@@ -38,13 +38,12 @@
...
     "prop-types": "^15.8.1",
-    "styled-components": "^4.4.1",
+    "styled-components": "^6.1.1",
     "styled-react-modal": "^3.1.1",
     "supercluster": "^8.0.1"
   },
   "peerDependencies": {
-    "react": "^16.14 || ^17",
-    "react-dom": "^16.14 || ^17"
+    "react": "^18"
```
Comment (ibgreen) [inline modules/react/package.json]: Let's follow deck.gl and use "^18.2.0": https://github.com/visgl/deck.gl/blob/master/modules/react/package.json#L42.

**Review by ibgreen — COMMENTED** [review]
@cg-sat Thanks for the PR. Can you also do a search for "react" dependencies across the repo and update across the board. In particular, can you make a pass on the examples directory?

**Comment by ibgreen** [comment]
As you can see in the log, you need to also commit the updated yarn.lock file when changing dependencies. Otherwise CI will reject the PR.

**Comment by ibgreen** [comment]
@cg-sat, looks like we have a merge conflict.

File: modules/react/package.json (hunk 1 above)
Comment (cg-sat) [reply modules/react/package.json]: I changed it to `^18.2.0`. Also updated to examples, that were still requesting react 16.

**Review by ibgreen — APPROVED** [review]
> I changed it to ^18.2.0. Also updated to examples, that were still requesting react 16.

Perfect, will merge as soon as clean

**Comment by cg-sat** [comment]
> @cg-sat, looks like we have a merge conflict.

Fixing it just now.

**Pull request merged**